okcupyd/profile.py

```diff
@@ -240,17 +254,28 @@ def enemy_percentage(self):
                   associated with this object.
         """
         return int(self._percentages_and_ratings_xpb.
-                   div.with_class('enemy').
-                   span.with_class('percent').
-                   get_text_(self.profile_tree).strip('%'))
+                   div.with_classes('matchgraph','matchgraph--enemy').
+                   div.with_class('matchgraph-graph').
+                   canvas.select_attribute_('data-pct').
+                   one_(self.profile_tree))
+
+    _location_xpb = xpb.span.with_class('userinfo2015-basics-asl-location')
+    _user_location_xpb = xpb.span(id='ajax_location')
 
-    _location_xpb = xpb.span(id='ajax_location')
     @util.cached_property
     def location(self):
         """
         :returns: The location of the user associated with this profile.
         """
-        return self._location_xpb.get_text_(self.profile_tree)
+
+        try:
+            #Attempt to get a non logged-in user's profile location
+            profile_location = self._location_xpb.get_text_(self.profile_tree)      
+        except:
```

Comment:
rewrote more elegantly, using self.is_logged_in_user, in https://github.com/kfred/okcupyd/commit/da49f95d1fdeddd57f96d82ced2050cd32c29bcb